packages/tools/src/stateManagement/annotation/config/getState.ts

```diff
@@ -7,7 +7,7 @@ import { AnnotationStyleStates } from '../../../enums';
  * Given a Annotation object, return the annotationStyle State that it
  * should be in based on its data
  * @param annotation - The annotation that we want to style.
- * @returns The state of the annotation whether it is Default, Highlighted, Locked, or Selected.
+ * @returns The state of the annotation whether it is Default, Highlighted, Locked, Selected, or AutoGenerated.
```

Comment:
What is the state diagram here?  That is, what happens if you then select an auto generated object - does it become highlighted?  What happens if you then deselect it?  How does it know to go to AutoGenerated and not Default?  What happens if you then lock an auto generated item?  
@sedghi  - is there a way to add styles for specific tools that take an extra parameter to use a different set of styles?  Perhaps using a different toolName would work?  That would then allow you to convert between autogenerated and regular ones, and choose whether or not you can edit it after auto generation without tweaking it?